local-cli/runAndroid/runAndroid.js

```diff
@@ -156,6 +156,20 @@ function buildAndRun(args) {
       'utf8'
     ).match(/package="(.+?)"/)[1];
 
+    const variantName = args.variant || 'debug';
+    const gradleFile = fs.readFileSync('app/build.gradle', 'utf8');
+
+    const appId = gradleFile.match(/applicationId "(.+?)"/)[1];
+    const buildTypes = gradleFile.substr(gradleFile.indexOf('buildTypes'));
+    const buildLoc = buildTypes.indexOf(variantName);
+    let appIdSuffix = '';
+    if (buildLoc >= 0) {
+        const buildSectionPlus = buildTypes.substr(buildLoc);
+        const buildSection = buildSectionPlus.substr(0, buildSectionPlus.indexOf('}'));
+        appIdSuffix = buildSection.match(/applicationIdSuffix "(.+?)"/)[1];
+    }
+
```

Comment:
Do we need to do heavy parsing here? Isn't it better to just allow appSuffix to be specified as an argument to a function? 

Also, if we make `appIdSuffix` an option that can be passed to a command, we can refer to it here as `args.appidSuffix` and benefit from the `default` property (described above in the previous comment). That `default` property can be also a function that is called when value has to be read (it's lazy & doesn't get called when custom appId is passed). We could do that `processing` inside that function.

I would strongly prefer this function to live in a separate file and be called `getAppIdFromGradle` so that we can put it inside `core/android` (old `rnpm`) and ideally reuse in other modules.